machine-runner/tests/esm/runner-types.test.ts

```diff
@@ -0,0 +1,319 @@
+import { describe, expect, it } from '@jest/globals'
+import {
+  createMachineRunner,
+  MachineEvent,
+  SwarmProtocol,
+  StateFactory,
+  State,
+} from '../../lib/esm/index.js'
+import { MachineRunner } from '../../lib/esm/runner/runner.js'
+import { NOP } from '../../lib/esm/utils/misc.js'
+import {
+  Equal,
+  Expect,
+  NotAnyOrUnknown,
+  NotEqual,
+  SerializableObject,
+  SerializableValue,
+} from '../../lib/esm/utils/type-utils.js'
+import * as ProtocolOneTwo from './protocol-one-two.js'
+import * as ProtocolScorecard from './protocol-scorecard.js'
+import { Runner } from './helper.js'
+
+/**
+ * In this particular test group, bad-good assertions are not required.
+ * This blocks only tests types by making type assignments.
+ * Bad type definitions are expected to fail the compilation
+ */
+describe('typings', () => {
+  // Type Tests
+  // ==========
+
+  /**
+   * This line guards the number of required parameters in MachineRunner
+   * It must be 2 until the next breaking change
+   */
+  type MachineRunnerIsCommonlyUsableTypes = MachineRunner<'SomeSwarmName', 'SomeMachineName'>
+
+  // Reusables
+  // ==========
+  const { Initial, Second } = ProtocolOneTwo
+
+  const E1 = MachineEvent.design('E1').withoutPayload()
+  const E2 = MachineEvent.design('E2').withoutPayload()
+  const E3 = MachineEvent.design('E3').withPayload<{ property: string }>()
+
+  const protocol = SwarmProtocol.make('example', [E1, E2])
+
+  it('event type transformation should be working well', () => {
+    true as Expect<Equal<MachineEvent.Of<typeof E1>, { type: 'E1' } & Record<never, never>>>
+    true as Expect<Equal<MachineEvent.Of<typeof E3>, { type: 'E3' } & { property: string }>>
+    true as Expect<
+      Equal<
+        MachineEvent.Factory.MapToActyxEvent<readonly [typeof E1, typeof E2, typeof E3]>,
+        [
+          ActyxEvent<MachineEvent.Of<typeof E1>>,
+          ActyxEvent<MachineEvent.Of<typeof E2>>,
+          ActyxEvent<MachineEvent.Of<typeof E3>>,
+        ]
+      >
+    >
+    true as Expect<
+      Equal<
+        MachineEvent.Factory.MapToMachineEvent<readonly [typeof E1, typeof E2, typeof E3]>,
+        [MachineEvent.Of<typeof E1>, MachineEvent.Of<typeof E2>, MachineEvent.Of<typeof E3>]
+      >
+    >
+    true as Expect<
+      Equal<
+        MachineEvent.Factory.MapToPayload<readonly [typeof E1, typeof E2, typeof E3]>,
+        [
+          MachineEvent.Payload.Of<typeof E1>,
+          MachineEvent.Payload.Of<typeof E2>,
+          MachineEvent.Payload.Of<typeof E3>,
+        ]
+      >
+    >
+    true as Expect<
+      Equal<
+        MachineEvent.Factory.Reduce<readonly [typeof E1, typeof E2, typeof E3]>,
+        typeof E1 | typeof E2 | typeof E3
+      >
+    >
+    true as Expect<
+      Equal<
+        MachineEvent.Of<MachineEvent.Factory.Reduce<readonly [typeof E1, typeof E2, typeof E3]>>,
+        MachineEvent.Of<typeof E1> | MachineEvent.Of<typeof E2> | MachineEvent.Of<typeof E3>
+      >
+    >
+  })
```

Comment:
not new, but still nice!

Reply:
It might not make sense but, splitting the test code was a really satisfying and relaxing experience.